src/index.ts

```diff
@@ -605,32 +605,15 @@ class ThreadPool {
       trackUnmanagedFds: this.options.trackUnmanagedFds,
     })
 
-    const { port1, port2 } = new MessageChannel()
-    const workerInfo = new WorkerInfo(worker, port1, onMessage)
-    if (this.startingUp) {
-      // There is no point in waiting for the initial set of Workers to indicate
-      // that they are ready, we just mark them as such from the start.
-      workerInfo.markAsReady()
-    }
-
-    const message: StartupMessage = {
-      filename: this.options.filename,
-      name: this.options.name,
-      port: port2,
-      sharedBuffer: workerInfo.sharedBuffer,
-      useAtomics: this.options.useAtomics,
-    }
-    worker.postMessage(message, [port2])
-
-    function onMessage(message: ResponseMessage) {
+    const onMessage = (message: ResponseMessage) => {
       const { taskId, result } = message
       // In case of success: Call the callback that was passed to `runTask`,
       // remove the `TaskInfo` associated with the Worker, which marks it as
       // free again.
       const taskInfo = workerInfo.taskInfos.get(taskId)
       workerInfo.taskInfos.delete(taskId)
 
-      pool.workers.maybeAvailable(workerInfo)
+      if (!this.options.isolateWorkers) pool.workers.maybeAvailable(workerInfo)
```

Comment:
This is the only line changed. The others are dum diff of moving a function to top

Reply:
Oh, this seems reasonable.